### PR TITLE
Checkreqs

### PR DESCRIPTION
**Pull request by Adrianzatreanu**
### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [ ] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [ ] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [ ] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3327/commits/efc0396c4edf5edc18a1ad6e7c26b9fe1f2769ec

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -323,14 +323,17 @@ def check_prerequisites(cls):
...
+        This function gets executed at construction.
 
         Section value requirements shall be checked inside the ``run`` method.
 
         :return: True if prerequisites are satisfied, else False or a string
                  that serves a more detailed description of what's missing.
         """
+        for requirement in cls.REQUIREMENTS:
+            if not requirement.is_installed():
+                print(requirement + " is not installed.")
```
Comment (sils) [inline coalib/bears/Bear.py]: return it

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3330

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3327/commits/87214670c4213448bfbeda830e501fb22e8234dd

File: requirements.txt [hunk 2]
```diff
@@ -5,3 +5,4 @@ colorlog~=2.7
 libclang-py3~=3.4.0
 Pygments~=2.1
 PyPrint~=0.2.6
+dependency_management~=0.1.6
```
Comment (Mixih) [inline requirements.txt]: Alphabetical order please ;)

**Comment by Mixih** [comment]
Travis should not a be a problem as circle works.. oh well :/

File: docs/Developers/Newcomers_Guide.rst [hunk 3]
```diff
@@ -109,7 +109,7 @@ strongly recommend following a tutorial, such as `this one <https://try.github.i
 
 We also have a page dedicated to git commands that will help you learn the
 basics,
-:doc:`here <Git_Basics>`.
+:doc:`on this page <Git_Basics>`.
```
Comment (Mixih) [inline docs/Developers/Newcomers_Guide.rst]: This seems to use "page" twice which seems a little grammatically strange to me. Why not make it an anonymous link?

**Comment by Mixih** [comment]
ack feaf2af

**Comment by Mixih** [comment]
screw travis ATM

**Comment by Mixih** [comment]
unack feaf2af

**Comment by Mixih** [comment]
doc problem

**Comment by Mixih** [comment]
ack ca9b5b2

**Comment by Mixih** [comment]
reack 76b0ad6 efc0396

**Comment by Mixih** [comment]
ack 76b0ad6

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Mixih** [comment]
ok , that's a screwy one

**Comment by Mixih** [comment]
@rultor merge

**Comment by Mixih** [comment]
@rultor merge

**Comment by Mixih** [comment]
Looks like rultor is timing out. We'll try again tomorrow

**Comment by Adrianzatreanu** [comment]
reack c7db0b6 3983d2e f64f32e

**Comment by Mixih** [comment]
Appveryor is being a meanie. Please force push

**Comment by Mixih** [comment]
reack 1ce238f

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
meh

**Comment by Mixih** [comment]
welp, let's try this again....

**Comment by Mixih** [comment]
@rultor merge

**Comment by Mixih** [comment]
ok *now* it's time to summon sils

/me summons @sils : Rultor issues
(and a jerk)

**Comment by Makman2** [comment]
@Mixih @Adrianzatreanu I was able to merge sth again, rebase and lgtm

**Comment by Adrianzatreanu** [comment]
repushed

**Comment by Adrianzatreanu** [comment]
reack 8552c09 c6b3c1e 16b8758

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
cmooooon. @sils , @Makman2 :( ?

**Comment by Mixih** [comment]
This is is hopeless... Rultor is being stupid here if all else fails, let's cherry-pick your commits to another branch.

**Comment by Mixih** [comment]
@Adrianzatreanu Willing to attempt one last try? If it still fails, try opening a new pr and cherry picking your commits off.

**Comment by Adrianzatreanu** [comment]
I threw the travis commit away, just trying to merge my stuff now, waiting to see if it fails now or not :)

**Comment by Adrianzatreanu** [comment]
travis.. ?

**Comment by jayvdb** [comment]
The problem with Travis is it tests what happens on 2.7 and 3.3, to ensure they fail to install.
Since `dependency_management` installs `coala`, and you are adding `dependency_management ` to `requirements.txt`, you are now specifying `coala` as an indirect requirements of `coala`.  Which is a bit bizarre.

IMO https://gitlab.com/coala/package_manager/issues/2 should be solved first, so we don't have package cross-dependencies, and that will magically solve the Travis problems here.

**Comment by Adrianzatreanu** [comment]
ah right, forgot bout it, solving the circular dependency now then :)

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/3327/commits/df65fd6f753b98e3dd5eef1a8297252115f3d6ff https://github.com/coala/coala/pull/3327/commits/746a730f8f211f6cf354695e8d5dac9839747ebe

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
cool, worked :)

**Pull request merged**